homeassistant/components/satel_integra/config_flow.py

```diff
@@ -0,0 +1,486 @@
+                        },
+                    }
+                )
+
+            for zone_number, zone_data in import_config.get(CONF_ZONES, {}).items():
+                subentries.append(
+                    {
+                        "subentry_type": SUBENTRY_TYPE_ZONE,
+                        "title": zone_data[CONF_NAME],
+                        "unique_id": f"{SUBENTRY_TYPE_ZONE}_{zone_number}",
+                        "data": {
+                            CONF_NAME: zone_data[CONF_NAME],
+                            CONF_ZONE_NUMBER: zone_number,
+                            CONF_ZONE_TYPE: zone_data.get(CONF_ZONE_TYPE, "motion"),
+                        },
+                    }
+                )
+
+            for output_number, output_data in import_config.get(
+                CONF_OUTPUTS, {}
+            ).items():
+                subentries.append(
+                    {
+                        "subentry_type": SUBENTRY_TYPE_OUTPUT,
+                        "title": output_data[CONF_NAME],
+                        "unique_id": f"{SUBENTRY_TYPE_OUTPUT}_{output_number}",
+                        "data": {
+                            CONF_NAME: output_data[CONF_NAME],
+                            CONF_OUTPUT_NUMBER: output_number,
+                            CONF_ZONE_TYPE: output_data.get(CONF_ZONE_TYPE, "motion"),
+                        },
+                    }
+                )
+
+            for switchable_output_number, switchable_output_data in import_config.get(
+                CONF_SWITCHABLE_OUTPUTS, {}
+            ).items():
+                subentries.append(
+                    {
+                        "subentry_type": SUBENTRY_TYPE_SWITCHABLE_OUTPUT,
+                        "title": switchable_output_data[CONF_NAME],
+                        "unique_id": f"{SUBENTRY_TYPE_SWITCHABLE_OUTPUT}_{switchable_output_number}",
+                        "data": {
+                            CONF_NAME: switchable_output_data[CONF_NAME],
+                            CONF_SWITCHABLE_OUTPUT_NUMBER: switchable_output_number,
+                        },
+                    }
+                )
+
+            return self.async_create_entry(
+                title=import_config[CONF_HOST],
+                data={
+                    CONF_HOST: import_config[CONF_HOST],
+                    CONF_PORT: import_config.get(CONF_PORT, DEFAULT_PORT),
+                },
+                options={CONF_CODE: import_config.get(CONF_CODE)},
+                subentries=subentries,
+            )
+
+        return self.async_abort(reason="Failed to connect")
+
+    async def test_connection(self, host, port) -> bool:
+        """Test a connection to the Satel alarm."""
+        controller = AsyncSatel(host, port, self.hass.loop)
+
+        result = await controller.connect()
+
+        # Make sure we close the connection again
+        controller.close()
+
+        return result
+
+
+class SatelOptionsFlow(OptionsFlow):
```

Comment:
I believe we now have a class that can automatically refresh the entry after updating options. I am not sure if the update listener in the async_setup_entry is meant for the new sub entries or the options flow

Reply:
There's still 1 option that is being used in the options flow, that might need to be reloaded, I think that's what the update listener was used for. I'll do some testing locally to see in what scenario's it triggers and if it's still needed.